fake_spectra/parallel.py

```diff
@@ -0,0 +1,85 @@
+import fake_spectra
+#print("fake_spectra imported")
+from fake_spectra.randspectra import RandSpectra
+#print("haloassigned_spectra imported")
+from fake_spectra.plot_spectra import PlottingSpectra
+#print("PlottingSpectra imported")
+from mpi4py import MPI
+import numpy as np
+
+comm = MPI.COMM_WORLD
+rank = comm.Get_rank()
+size = comm.Get_size()
+
+numlos = 1000
+ndla = 1000
+thresh_t = 10**20.3
+### the array to store added column density in
+
+
+
+if rank ==0 :
+    
+    found = 0
+
+    while (found < ndla) :
+
+        col_den_added = np.zeros(shape=(numlos,))
+
+        ### Recieve col_den from all other ranks and add them together
+        comm.Reduce(np.zeros(shape=(numlos,), dtype='d'), col_den_added, op = MPI.SUM, root = 0)
+        
+        ### indices which should be replaced
+        ind = np.where(col_den_added > thresh_t)
+        size_ind = np.size(ind)
+        found += size_ind
+        print('found = ', found)
+        ### Here, Broadcast not_DLA_indices to each rank to be regenrated
+
+
+        size_ind = comm.bcast(size_ind, root=0)
+        comm.Bcast(ind, root = 0)
```

Comment:
This broadcast too: it needs to be called on the other ranks so that the other ranks know when to receive the data.

Reply:
Yes, I see your point. As I explained about comm.Replace() in above comment, I have called comm.Bcast(ind , root = 0) in replace_not_DLA().
For labor ranks, would there be any difference if I call comm.Bcast(ind, root= 0)  in parallel.py instead of replace_not_DLA() ?